app/src/main/java/com/github/ashutoshgngwr/noice/WakeUpTimerManager.kt

```diff
@@ -141,6 +141,48 @@ object WakeUpTimerManager {
     }
   }
 
+  fun getStringScheduleLeftTime(context: Context, diffHours: Int, diffMinutes: Int): String {
+    return when (diffHours) {
+      0 -> {
+        when (diffMinutes) {
+          0 -> context.getString(R.string.wake_up_timer_schedule_set_minute, diffMinutes)
+          1 -> context.getString(R.string.wake_up_timer_schedule_set_minute, diffMinutes)
+          else -> context.getString(R.string.wake_up_timer_schedule_set_minutes, diffMinutes)
+        }
+      }
+      1 -> {
+        when (diffMinutes) {
+          0 -> context.getString(R.string.wake_up_timer_schedule_set_hour, diffHours)
+          1 -> context.getString(
+            R.string.wake_up_timer_schedule_set_hour_minute,
+            diffHours,
+            diffMinutes
+          )
+          else -> context.getString(
+            R.string.wake_up_timer_schedule_set_hour_minutes,
+            diffHours,
+            diffMinutes
+          )
+        }
+      }
+      else -> {
+        when (diffMinutes) {
+          0 -> context.getString(R.string.wake_up_timer_schedule_set_hours, diffHours)
+          1 -> context.getString(
+            R.string.wake_up_timer_schedule_set_hours_minute,
+            diffHours,
+            diffMinutes
+          )
+          else -> context.getString(
+            R.string.wake_up_timer_schedule_set_hours_minutes,
+            diffHours,
+            diffMinutes
+          )
+        }
+      }
+    }
+  }
```

Comment:
This isn't needed if you switch to [Plurals](https://developer.android.com/guide/topics/resources/string-resource#Plurals). But plurals don't support multiple params so you'll have to concatenate strings together. Both ways are hell to translate. Your call.

Plural string should look something like:

```xml
<string name="...">Alarm will go off in %1$s %2$s.</string><!-- but, you can't add 'and' now -->
<plurals name="hours">
  <item quantity="zero"></item>
  <item quantity="one">%d hour</item>
  <item quantity="other">%d hours</item>
</plurals>
<plurals name="minutes">
  <item quantity="zero"></item>
  <item quantity="one">%d minute</item>
  <item quantity="other">%d minutes</item>
</plurals>
```

The code is just for demo. It may not work as is.

Reply:
It's the first time I heard about plurals and this is a much better solution when it comes to reading code. So after playing with plurals I've decided to keep it and I'll update unit tests. Tell me which solution do you preffer once I update changes.